src/OSPSuite.Infrastructure.Import/Core/DataSheetCollection.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using OSPSuite.Utility.Collections;
+
+namespace OSPSuite.Infrastructure.Import.Core
+{
+   /// <summary>
+   ///    a collection of excel dataSheet sheets
+   /// </summary>
+   public class DataSheetCollection : IEnumerable<DataSheet>
+   {
+      private readonly Cache<string, DataSheet> _dataSheets;
+
+      public DataSheetCollection()
+      {
+         _dataSheets = new Cache<string, DataSheet>(x => x.SheetName);
```

Comment:
initialize where you define
private readonly Cache<string, DataSheet> _dataSheets = new Cache<string, DataSheet>(x => x.SheetName);

Reply:
done